modules/programs/gh.nix

```
@@ -126,7 +126,25 @@ in {
     home.packages = [ cfg.package ];
 
     xdg.configFile."gh/config.yml".source =
-      yamlFormat.generate "gh-config.yml" cfg.settings;
+      yamlFormat.generate "gh-config.yml" ({ version = "1"; } // cfg.settings);
+
+    # Version 2.40.0+ of gh needs to migrate account formats, this needs to
+    # happen before the version = 1 is placed in the configuration file. Running
+    # `gh help` is sufficient to perform the migration. If the migration already
+    # has occurred, then this is a no-op.
+    #
+    # See https://github.com/nix-community/home-manager/issues/4744 for details.
+    home.activation.migrateGhAccounts =
+      hm.dag.entryBetween [ "linkGeneration" ] [ "writeBoundary" ] ''
+        if [[ -e "${config.xdg.configHome}/gh/config.yml" ]]; then
+          TMP_DIR=$(mktemp -d)
+          cp "${config.xdg.configHome}/gh/hosts.yml" $TMP_DIR/
+          export GH_CONFIG_DIR=$TMP_DIR
+          $DRY_RUN_CMD ${getExe cfg.package} help 2>&1 > $DRY_RUN_NULL
+          cp $TMP_DIR/hosts.yml "${config.xdg.configHome}/gh/hosts.yml"
+          rm -rf $TMP_DIR
```

Replace the final `rm` command with the more reliable `trap` alternative:

```suggestion
          TMP_DIR=$(mktemp -d)
          trap "rm --force --recursive $TMP_DIR" EXIT
          cp "${config.xdg.configHome}/gh/hosts.yml" $TMP_DIR/
          export GH_CONFIG_DIR=$TMP_DIR
          $DRY_RUN_CMD ${getExe cfg.package} help 2>&1 > $DRY_RUN_NULL
          cp $TMP_DIR/hosts.yml "${config.xdg.configHome}/gh/hosts.yml"
```

The main advantage is that the temporary file is removed even when the programs exits in unexpected ways, as long as the `EXIT` event is not somehow circumvented.

Note that, I did not test this patch.

---

Change looks okay, I'll fold it into my patch once I'm back at my desktop to test it.